src/tools/web/client/src/components/TreeItem.jsx

```diff
@@ -14,48 +14,139 @@ import React from 'react'
 import TextField from 'material-ui/TextField'
 import IconButton from 'material-ui/IconButton'
 import ActionDelete from 'material-ui/svg-icons/action/delete'
+import ContentCreate from 'material-ui/svg-icons/content/create'
 
 import TreeView from './TreeView.jsx'
 
+// TODO: set min/max appropriately for these types
+const INTEGER_TYPES = [
+  'short', 'unsigned_short', 'long', 'unsigned_long', 'long_long',
+  'unsigned_long_long',
+]
+
+const FLOAT_TYPES = [ 'float', 'double' ]
+
+const isNumber = (value) => !isNaN(value)
+const isInt = (value) => /^(-|\+)?[0-9]+$/.test(value)
```

Comment:
Isn't there a better way to check this? (try to convert to int?)

It seems to allow 0 prefix.